src/lib.rs

```diff
@@ -182,50 +182,69 @@ pub mod mock;
 //
 // can't use both sync and async secret service
 //
-#[cfg(all(feature = "sync-secret-service", feature = "async-secret-service"))]
+#[cfg(any(
+    all(feature = "sync-secret-service", feature = "async-secret-service"),
+    all(
+        feature = "sync-persistent-keyutils",
+        feature = "async-persistent-keyutils",
+    )
+))]
 compile_error!("This crate cannot use the secret-service both synchronously and asynchronously");
```

Comment:
How about "This crate cannot use both the sync and async versions of any credential store"